src/uaclient/uasession.cpp

```diff
@@ -328,5 +328,69 @@ UaStatus UaSession::call(
 
 }
 
+UaStatus UaSession::browse(
+        ServiceSettings&         serviceSettings,
+        const UaNodeId&          nodeToBrowse,
+        const BrowseContext&     browseContext,
+        UaByteString&            continuationPoint,
+        UaReferenceDescriptions& referenceDescriptions)
+{
+	referenceDescriptions.clear();
+
+    UA_BrowseRequest browseRequest;
+    UA_BrowseRequest_init(&browseRequest);
+
+    browseRequest.nodesToBrowseSize = 1;
+    browseRequest.nodesToBrowse = UA_BrowseDescription_new();
+    nodeToBrowse.copyTo(&browseRequest.nodesToBrowse[0].nodeId);
+    browseRequest.nodesToBrowse[0].browseDirection = UA_BROWSEDIRECTION_FORWARD; // TODO!! Important!! This setting should come from BrowseContext! It might be an oversimplificatrion.
+    browseRequest.nodesToBrowse[0].resultMask = UA_BROWSERESULTMASK_ALL;
+
+    UA_BrowseResponse browseResponse;
+    UA_BrowseResponse_init(&browseResponse);
+
+    browseResponse = UA_Client_Service_browse(m_client, browseRequest);
+
+    UaStatus finalStatusCode = browseResponse.responseHeader.serviceResult;
+
+    if (UaStatus(browseResponse.responseHeader.serviceResult).isGood())
```

Comment:
might be nice to LOG status if service result is not good - failure result code might one day help shine some light on the root problems for some unhealthy environment

Reply:
Fair point, will do.